### PR TITLE
Add test(s) that validate Pyright can talk over LSP

### LINKED ISSUE #6677: Provide tests at the LSP level in Pyright
Pylance has a series of tests we run on submission that run the language server in a worker thread and then exercise the language server by passing messages to the worker thread. 

This request is to add these sort of tests to Pyright as well. Theoretically this would have caught this issue:
#6676

### PR DESCRIPTION
**Pull request by rchiodo**
Fixes #6677 

This should make sure we don't break the background thread in any submissions.

### CODE REVIEW

File: packages/pyright-internal/src/tests/lsp/main.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/*
+ * node.ts
```
Comment (erictraut) [inline packages/pyright-internal/src/tests/lsp/main.ts:2]: File name is wrong in comment.

File: packages/pyright-internal/tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,9 @@
 {
     "extends": "../../tsconfig.json",
     "compilerOptions": {
-        "outDir": "./out"
+        "outDir": "./out",
+        "paths": {
```
Comment (erictraut) [inline packages/pyright-internal/tsconfig.json:5]: Is this change needed?

**Review by erictraut — APPROVED** [review]
Thanks for writing these tests. This is a good addition to the test suite.

Looks good overall. Just a few nits for code style conformance.

File: packages/pyright-internal/tsconfig.json (hunk 2 above)
Comment (rchiodo) [reply packages/pyright-internal/tsconfig.json:5]: Yeah otherwise tsconfigResolveAliases throws an exception.

File: packages/pyright-internal/src/tests/languageServer.test.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
+import assert from 'assert';
```
[the remark this answers is not in the document]
Comment (rchiodo) [reply packages/pyright-internal/src/tests/languageServer.test.ts:9]: Will do

File: packages/pyright-internal/src/tests/lsp/main.ts (hunk 1 above)
Comment (rchiodo) [reply packages/pyright-internal/src/tests/lsp/main.ts]: Will do

**Review by Loki419420 — COMMENTED** [review]
#7179
